eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -645,4 +645,9 @@ object Peer {
   }
 
   def hostAndPort2InetSocketAddress(hostAndPort: HostAndPort): InetSocketAddress = new InetSocketAddress(hostAndPort.getHost, hostAndPort.getPort)
+
+  /**
+    * Exponential backoff retry with a finite max
+    */
+  def nextReconnectionDelay(currentDelay: FiniteDuration): FiniteDuration = (2 * currentDelay).max(MAX_RECONNECT_INTERVAL)
```

Comment:
That's not really exponential but a linear increment of the delay, it'd suggest something like `Math.pow(2, currentDelay).max(MAX_RECONNECT_INTERVAL)`

Reply:
Okay on a second thought this wouldn't work because we need to store the number of attempts if we are going to do the exponential backoff.